README.md
# calculadora_prestamos_GUI

# Calculadora de Préstamos con interfaz gráfica : Versión 0.3


La Calculadora de Préstamos es una herramienta monousuario diseñada para su uso personal y profesional en entornos Windows y Linux. Proporciona estimaciones aproximadas de cuotas mensual o anual basadas en el sistema francés de interés compuesto.

Fecha de publicación: Julio de 2023
Autor: Juan Fuente
Edición: GRATUITA


## Captura de pantalla
![CapturaPrestamosGUI](https://github.com/Juan-Fuente-T/calculadora_prestamos_GUI/assets/127140423/28c9437a-2e13-466d-b5ca-0c8453c54e73)



## Contenido de este documento
 - Breve descripción de la aplicación
 - Novedades y características especiales en esta versión
 - Instrucciones de uso
 - Requisitos de uso
 - Créditos e información de contacto

### 1. Breve descripción de la aplicación
La Calculadora de Préstamos es una herramienta diseñada para calcular cuotas anuales y mensuales simulando un préstamo real. Permite ajustar el capital, el interés y el plazo de amortización.

### 2. Novedades y características especiales en esta versión
La Calculadora de Préstamos, versión 0.6, fue creada en julio de 2023 como parte del proyecto de estudio del curso IFCT0609, Desarrollo de Aplicaciones Informáticas. Esta versión representa el primer lanzamiento público, distribuido de manera gratuita. La intención detrás de esta aplicación es brindar una herramienta funcional mientras se aprende el desarrollo de software.

### 3. Instrucciones de uso
Para utilizar la Calculadora de Préstamos, siga estos pasos:

 - Ingrese la cantidad de capital a solicitar.
 - Ingrese el porcentaje de interés a aplicar.
 - Ingrese la duración en años como plazo de amortización.
 - Presione el botón **Calcular** para obtener los resultados.
La calculadora mostrará tanto la cuota mensual como la cuota anual. Los cálculos se basan en la fórmula real de interés compuesto del sistema francés.

### 4. Requisitos de uso
Antes de ejecutar el programa, asegúrate de tener instalado Python en tu sistema. Puedes hacerlo siguiendo estos pasos:

 -  Clona este repositorio en tu máquina local:
### git clone https://github.com/tu-usuario/tu-proyecto.git

 -  Ingresa al directorio del proyecto:
### cd tu-proyecto

 -  Instala las dependencias necesarias ejecutando el siguiente comando:
### pip install -r requirements.txt

 - Ejecuta el siguiente comando para iniciar la aplicación:
### python calculadora_prestamos_grafica-0.0.1.py (o adaptar al nombre exacto del archivo si es necesario)

### 5. Créditos e información de contacto

- ### Licencia
Este proyecto se distribuye bajo la licencia MIT. Puedes ver los detalles de la licencia aqui https://es.wikipedia.org/wiki/Licencia_MIT.

- ### Contacto
Si tienes alguna pregunta, sugerencia o comentario sobre este proyecto, no dudes en contactarme a través de los canales que tengo disponibles en mi web ' https://juanfuente.ovh '

## ¡Espero que disfrutes utilizando y te sea útil la calculadora de préstamos!

**© 2023 Juan Fuente**




changelog.txt
# *** Calculadora_prestamos_grafica :: Version 0.0.3 ***
# Creative Commons: Juan Fuente 2023

"""
Esta aplicación de interfaz gráfica, multiplataforma (Windows, Linux, Mac), ofrece la funcionalidad de cálculo de cuotas, mensuales y anuales, basada en el método francés de interés compuesto. Toma como valores el capital, el interés a aplicar y el plazo en años para la devolución para calcular ambos valores. 


Novedades de la versión 0.0.3
- Se añaden la funcionalidad de que se borren las cantidades al entrar en foco después de haber calculado resultado al entrar en foco en las ventanas de toma de datos
- Se termina la documentacion
- Se eliminan lineas no necesarias y comentarios

Novedades de la versión 0.0.2
- Se añaden mensajes de error en caso de introducir valores incorrectos
- Se borra la pantalla del valor que se ha introducido si se introduce un valor incorrecto
- Se borran las pantallas de resultados si se introduce un valor incorrecto
- Se colocan en su lugar todas las ventanas y pantallas de resultados
- Se añaden todos los colores de la aplicación
- Se añade el icono




calculadora_prestamos_grafica-0.0.2.py
# *** Calculadora de prestamos  :: Version 0.0.3 ***

# ©2023 Calculadora financiera de Juan Fuente

"""
Esta aplicación multiplataforma (Windows, Linux, Mac) ofrece una calculadora de prestamos utilizando el interes frances que opera mediante interfaz gráfica


Novedades de la versión 0.0.2
- Se añaden mensajes de error en caso de introducir valores incorrectos
- Se borra la pantalla del valor que se ha introducido si se introduce un valor incorrecto
- Se borran las pantallas de resultados si se introduce un valor incorrecto
- Se colocan en su lugar todas las ventanas y pantallas de resultados
- Se añaden todos los colores de la aplicación
- Se añade el icono


"""


import tkinter as tk
from tkinter import messagebox  #se importa messagebox para mostrar mensaje de error
from tkinter import ttk #ttk es necesario para los desplegables combobox 


class calculadora_prestamos(tk.Tk): 

    def __init__(self): 
        super().__init__()

        self.title("Calculadora de préstamos") #titulo de la aplicacion
        self.iconbitmap("calculadora_prestamo.ico") #icono de la aplicacion
        self.geometry("550x700+100+100") #tamaño de la aplicacion
        self.configure(bg="#FFDDB0") #color de fondo de la aplicacion

        self.pantalla_cuota_anual = tk.Entry(self, width=10, justify="right", borderwidth=2, fg="#191007", bg="#CFB38F") #se define la pantalla donde se va a mostrar el resultado 
        self.pantalla_cuota_mensual = tk.Entry(self, width=10, justify="right", borderwidth=2, fg="#191007", bg="#CFB38F") #se define la pantalla donde se va a mostrar el resultado 
        
        self.duracion_prestamo = tk.Entry(self, width=12, justify="right", borderwidth=2, fg="#191007", bg="#CFB38F") #se define la ventana donde se van a introducir la cantidad de moneda
        self.capital = tk.Entry(self, width=12, justify="right", borderwidth=2, fg="#191007", bg="#CFB38F")#se define el desplegable para moneda origen
        #self.interes = tk.Entry(self, width=15, justify="right", borderwidth=2, fg="navy", bg="LightSteelBlue1")#se define el desplegable para moneda origen
        self.interes = tk.Entry(self, width=12, justify="right", borderwidth=2, fg="#191007", bg="#CFB38F")#se define el desplegable para la moneda destino
        
        self.calcular_button = tk.Button(self, text="Calcular", fg="bisque", bg="#82715A", padx=40, pady=4, command=self.mostrar_cuota) #se define el boton para realizar la conversion


        self.pantalla_cuota_anual.place(x=430, y=100, anchor="center") #se define la colocacion de la patalla de resultados
        self.pantalla_cuota_mensual.place(x=430, y=200, anchor="center") #se define la colocacion de la patalla de resultados
        
        self.duracion_prestamo.place(x=130, y=270, anchor="center")#se define la colocacion de la ventana de cantidad de  moneda 
        self.capital.place(x=130, y=70, anchor="center")#se define la colocacion de la ventana de moneda origen
        self.interes.place(x=130, y=170, anchor="center")#se define la colocacion de la ventana de moneda destino
        
        self.calcular_button.place(x=430, y=350, anchor="center")#se define la colocacion del boton de convertir
        

        self.pantalla_cuota_anual.configure(font=("Arial Black", 18))#se configura el tipo de letra y tamaño de la ventana de cantidad de  moneda 
        self.pantalla_cuota_mensual.configure(font=("Arial Black", 18))#se configura el tipo de letra y tamaño de la ventana de cantidad de  moneda 
        self.duracion_prestamo.configure(font=("Arial Black", 18))#se configura el tipo de letra y tamaño de la ventana de cantidad de  moneda 
        self.calcular_button.configure(font=("Arial", 14))#se configura el tipo de letra y tamaño del boton de convertir
        self.capital.configure(font=("Arial Black", 18))#se configura el tipo de letra y tamaño de la ventana de cantidad de  moneda 
        self.interes.configure(font=("Arial Black", 18))#se configura el tipo de letra y tamaño de la ventana de cantidad de  moneda 


        etiqueta_cuota_anual= tk.Label(self, text="Cuota anual:", fg="#191007",bg="#FFC06E", font=("Arial", 13))#se define la etiqueta de la pantalla de resultados
        etiqueta_cuota_anual.place(x=430, y=50, anchor="center")#se define la colocacion de la etiqueta de la pantalla de resultado
        etiqueta_cuota_mensual= tk.Label(self, text="Cuota mensual:", fg="#191007",bg="#FFC06E", font=("Arial", 13))#se define la etiqueta de la pantalla de resultados
        etiqueta_cuota_mensual.place(x=430, y=150, anchor="center")#se define la colocacion de la etiqueta de la pantalla de resultado
        

        duracion_prestamo_label = tk.Label(self, text="Plazo en años:", fg="#191007",bg="#82715A", font=("Arial", 13) )#se define la etiqueta de cantidad de  moneda 
        duracion_prestamo_label.place(x=130, y=230, anchor="center")#se define la colocacion de la etiqueta de cantidad moneda 
        

        capital_label = tk.Label(self, text="Capital:", fg="#191007",bg="#82715A", font=("Arial", 13))#se define la etiqueta de moneda origen
        capital_label.place(x=130, y=30, anchor="center")#se define la colocacion de la etiqueta de moneda origen

        interes_label = tk.Label(self, text="Interés:", fg="#191007",bg="#82715A", font=("Arial", 13))#se define la etiqueta de moneda destino
        interes_label.place(x=130, y=130, anchor="center") #se define la colocacion de la etiqueta de moneda destino
        calcular_button_label = tk.Label(self, text=""" 
    Calculadora de préstamos
                    
Introduzca la cantidad de capital, el interés a aplicar 
y el número de años para su amortización 
Con el botón Calcular obtendrá la cuota mensual 
              y la cuota anual resultantes.                                   

        CC-BY-NC 2023 Juan Fuente         
                        """, fg="#191007", bg="#82715A", font=("Arial", 13),)
        calcular_button_label.place(x=275, y=520, anchor="center", width= 500) #se define la colocacion de la etiqueta bajo el boton de convertir
        

        

    def calcular_cuota(self):
        #cantidad = self.duracion_prestamo.get()
        capital = self.capital.get()
        interes = self.interes.get()
        duracion_prestamo = self.duracion_prestamo.get()

        if not capital.replace('.', '', 1).isdigit():
            self.capital.delete(0, tk.END)
            self.pantalla_cuota_mensual.delete(0, tk.END)
            self.pantalla_cuota_anual.delete(0, tk.END)
            messagebox.showerror("Error", "El valor del capital no es válido")
            return
        if not interes.replace('.', '', 1).isdigit():
            self.interes.delete(0, tk.END)
            self.pantalla_cuota_mensual.delete(0, tk.END)
            self.pantalla_cuota_anual.delete(0, tk.END)
            messagebox.showerror("Error", "El valor del interés no es válido")
            return
        #if not 1 <= float(duracion_prestamo) <= 35 or not duracion_prestamo.isdigit():
        if not duracion_prestamo.isdigit():

            self.duracion_prestamo.delete(0, tk.END)
            self.pantalla_cuota_mensual.delete(0, tk.END)
            self.pantalla_cuota_anual.delete(0, tk.END)
            messagebox.showerror("Error", "El valor de la duración del préstamo no es válido")
            return
        
        duracion_meses = float(duracion_prestamo) * 12
        capital, interes, duracion_prestamo = float(capital), float(interes), float(duracion_prestamo)
        cuota_anual = round(((capital * interes / 100) / (1 - (1 + interes / 100) ** -duracion_prestamo)), 2)
        cuota_mensual = round(((capital * interes / 1200) / (1 - (1 + interes / 1200) ** -duracion_meses)), 2)
        return cuota_anual, cuota_mensual
        """except ValueError:
            
            return None, None"""

        
        """ if not nombre_moneda_origen or not nombre_moneda_destino:
            if event is None or (event.widget == self.calcular_button if event else False):
                tk.messagebox.showerror("Error", "Por favor, seleccione la moneda de origen y la moneda de destino.")
                return
"""

    def mostrar_cuota(self):  # New function to handle the calculation and display
        cuota_anual,cuota_mensual = self.calcular_cuota()
        
    
        self.pantalla_cuota_anual.delete(0, tk.END)
        self.pantalla_cuota_anual.insert(tk.END, cuota_anual)

        self.pantalla_cuota_mensual.delete(0, tk.END)
        self.pantalla_cuota_mensual.insert(tk.END, cuota_mensual)
        

if __name__ == "__main__":
    ventana = calculadora_prestamos()
    ventana.mainloop()

grafica/calculadora_prestamos_grafica-0.0.1.py
import tkinter as tk
from tkinter import messagebox  #se importa messagebox para mostrar mensaje de error
from tkinter import ttk #ttk es necesario para los desplegables combobox 


class calculadora_prestamos(tk.Tk): 

    def __init__(self): 
        super().__init__()

        self.title("Calculadora de préstamos") #titulo de la aplicacion
        self.iconbitmap("conversor_divisas.ico") #icono de la aplicacion
        self.geometry("550x700+100+100") #tamaño de la aplicacion
        self.configure(bg="AntiqueWhite4") #color de fondo de la aplicacion

        self.pantalla_cuota_anual = tk.Entry(self, width=12, borderwidth=2, fg="navy", bg="LightSteelBlue1") #se define la pantalla donde se va a mostrar el resultado 
        self.pantalla_cuota_mensual = tk.Entry(self, width=12, borderwidth=2, fg="navy", bg="LightSteelBlue1") #se define la pantalla donde se va a mostrar el resultado 
        self.duracion_prestamo = tk.Entry(self, width=12, borderwidth=2, fg="navy", bg="LightSteelBlue1") #se define la ventana donde se van a introducir la cantidad de moneda
        
        self.capital = tk.Entry(self, width=12, borderwidth=2, fg="navy", bg="LightSteelBlue1")#se define el desplegable para moneda origen
        self.interes = tk.Entry(self, width=12, borderwidth=2, fg="navy", bg="LightSteelBlue1")#se define el desplegable para la moneda destino
        self.calcular_button = tk.Button(self, text="Calcular", fg="navy", bg="LightSteelBlue1", padx=40, pady=4, command=self.mostrar_cuota) #se define el boton para realizar la conversion

        self.pantalla_cuota_anual.place(x=480, y=120, anchor="center") #se define la colocacion de la patalla de resultados
        self.pantalla_cuota_mensual.place(x=480, y=180, anchor="center") #se define la colocacion de la patalla de resultados
        self.duracion_prestamo.place(x=120, y=180, anchor="center")#se define la colocacion de la ventana de cantidad de  moneda 
        self.capital.place(x=130, y=70, anchor="center")#se define la colocacion de la ventana de moneda origen
        self.interes.place(x=470, y=70, anchor="center")#se define la colocacion de la ventana de moneda destino
        self.calcular_button.place(x=300, y=265, anchor="center")#se define la colocacion del boton de convertir
        

        self.pantalla_cuota_anual.configure(font=("Arial Black", 18))#se configura el tipo de letra y tamaño de la ventana de cantidad de  moneda 
        self.pantalla_cuota_mensual.configure(font=("Arial Black", 18))#se configura el tipo de letra y tamaño de la ventana de cantidad de  moneda 
        self.duracion_prestamo.configure(font=("Arial Black", 18))#se configura el tipo de letra y tamaño de la ventana de cantidad de  moneda 
        self.calcular_button.configure(font=("Arial", 14))#se configura el tipo de letra y tamaño del boton de convertir
        self.capital.configure(font=("Arial", 18))#se configura el tipo de letra y tamaño de la ventana de cantidad de  moneda 
        self.interes.configure(font=("Arial", 18))#se configura el tipo de letra y tamaño de la ventana de cantidad de  moneda 


        etiqueta_cuota_anual= tk.Label(self, text="Cuota anual:", fg="navy",bg="LightSteelBlue1", font=("Arial", 13))#se define la etiqueta de la pantalla de resultados
        etiqueta_cuota_anual.place(x=480, y=120, anchor="center")#se define la colocacion de la etiqueta de la pantalla de resultado
        etiqueta_cuota_mensual= tk.Label(self, text="Cuota mensual:", fg="navy",bg="LightSteelBlue1", font=("Arial", 13))#se define la etiqueta de la pantalla de resultados
        etiqueta_cuota_mensual.place(x=480, y=160, anchor="center")#se define la colocacion de la etiqueta de la pantalla de resultado
        

        duracion_prestamo_label = tk.Label(self, text="Plazo en años:", fg="navy",bg="LightSteelBlue1", font=("Arial", 13) )#se define la etiqueta de cantidad de  moneda 
        duracion_prestamo_label.place(x=120, y=120, anchor="center")#se define la colocacion de la etiqueta de cantidad moneda 
        

        capital_label = tk.Label(self, text="Capital:", fg="navy",bg="LightSteelBlue1", font=("Arial", 13))#se define la etiqueta de moneda origen
        capital_label.place(x=120, y=30, anchor="center")#se define la colocacion de la etiqueta de moneda origen

        interes_label = tk.Label(self, text="Interés:", fg="navy",bg="LightSteelBlue1", font=("Arial", 13))#se define la etiqueta de moneda destino
        interes_label.place(x=480, y=30, anchor="center") #se define la colocacion de la etiqueta de moneda destino
        calcular_button_label = tk.Label(self, text=""" 
Calculadora de préstamos
                    
Introduzca la cantidad de capital, el interés a aplicar 
y el número de años para su amortización 
y al pulsar el botón de calcular
obtendrá la cuota mensual y anual resultantes.                                   

        CC-BY-NC 2023 Juan Fuente         
                        """, fg="navy", bg="LightSteelBlue1", font=("Arial", 13))
        calcular_button_label.place(x=300, y=440, anchor="center") #se define la colocacion de la etiqueta bajo el boton de convertir
        

        

    def calcular_cuota(self):
        #cantidad = self.duracion_prestamo.get()
        capital = self.capital.get()
        interes = self.interes.get()
        duracion_prestamo = self.duracion_prestamo.get()
        duracion_meses = float(duracion_prestamo) * 12


        try:
            capital, interes, duracion_prestamo = float(capital), float(interes), float(duracion_prestamo)
        except ValueError:
            self.pantalla_cuota_anual.delete(0, tk.END)
            return
        cuota_anual = (capital * interes/100) / (1 - (1 + interes/100)** -duracion_prestamo)
        cuota_mensual = (capital * interes/1200) / (1 - (1 + interes/1200) ** -duracion_meses)
        return cuota_anual, cuota_mensual
        """#funcion para calcular la anualidad
        def calcular_anualidad(capital, interes, años): #toma tres argumentos y devueve el resultado
            return (capital * interes/100) / (1 - (1 + interes/100)** -años)
        #funcion para calcular la mensualidad
        def calcular_mensualidad(capital, interes, meses):#toma tres argumentos y devuelve el resultado
            return (capital * interes/1200) / (1 - (1 + interes/1200) ** -meses)"""

    def mostrar_cuota(self):  # New function to handle the calculation and display
        cuota_anual = self.calcular_cuota()
        cuota_mensual = self.calcular_cuota()
        
        self.pantalla_cuota_anual.delete(0, tk.END)
        self.pantalla_cuota_anual.insert(tk.END, cuota_anual)
    
        self.pantalla_cuota_mensual.delete(0, tk.END)
        self.pantalla_cuota_mensual.insert(tk.END, cuota_mensual)
    

if __name__ == "__main__":
    ventana = calculadora_prestamos()
    ventana.mainloop()

calculadora_prestamos_grafica-0.0.3.py
# *** Calculadora de prestamos  :: Version 0.0.3 ***

# ©2023 Calculadora financiera de Juan Fuente

"""
Esta aplicación multiplataforma (Windows, Linux, Mac) ofrece una calculadora de prestamos utilizando el interes frances que opera mediante interfaz gráfica

Novedades de la versión 0.0.3

- Se añaden la funcionalidad de que se borren las cantidades al entrar en foco después de haber calculado resultado al entrar en foco en las ventanas de toma de datos
- Se termina la documentacion
- Se eliminan lineas no necesarias y comentarios


Novedades de la versión 0.0.2
- Se añaden mensajes de error en caso de introducir valores incorrectos
- Se borra la pantalla del valor que se ha introducido si se introduce un valor incorrecto
- Se borran las pantallas de resultados si se introduce un valor incorrecto
- Se colocan en su lugar todas las ventanas y pantallas de resultados
- Se añaden todos los colores de la aplicación
- Se añade el icono


"""


import tkinter as tk
from tkinter import messagebox  #se importa messagebox para mostrar mensaje de error
from tkinter import ttk #ttk es necesario para los desplegables combobox 


class calculadora_prestamos(tk.Tk): 

    def __init__(self): 
        super().__init__()

        self.title("Calculadora de préstamos") #titulo de la aplicacion
        self.iconbitmap("calculadora_prestamo.ico") #icono de la aplicacion
        self.geometry("550x700+100+100") #tamaño de la aplicacion
        self.configure(bg="#FFDDB0") #color de fondo de la aplicacion

        self.pantalla_cuota_anual = tk.Entry(self, width=10, justify="right", borderwidth=2, fg="#191007", bg="#CFB38F") #se define la pantalla donde se va a mostrar el resultado 
        self.pantalla_cuota_mensual = tk.Entry(self, width=10, justify="right", borderwidth=2, fg="#191007", bg="#CFB38F") #se define la pantalla donde se va a mostrar el resultado 
        
        self.duracion_prestamo = tk.Entry(self, width=12, justify="right", borderwidth=2, fg="#191007", bg="#CFB38F") #se define la ventana donde se van a introducir la cantidad de moneda
        self.capital = tk.Entry(self, width=12, justify="right", borderwidth=2, fg="#191007", bg="#CFB38F")#se define el desplegable para moneda origen
        #self.interes = tk.Entry(self, width=15, justify="right", borderwidth=2, fg="navy", bg="LightSteelBlue1")#se define el desplegable para moneda origen
        self.interes = tk.Entry(self, width=12, justify="right", borderwidth=2, fg="#191007", bg="#CFB38F")#se define el desplegable para la moneda destino
        
        self.calcular_button = tk.Button(self, text="Calcular", fg="bisque", bg="#82715A", padx=40, pady=4, command=self.mostrar_cuota) #se define el boton para realizar la conversion


        self.pantalla_cuota_anual.place(x=430, y=100, anchor="center") #se define la colocacion de la patalla de resultados
        self.pantalla_cuota_mensual.place(x=430, y=200, anchor="center") #se define la colocacion de la patalla de resultados
        
        self.duracion_prestamo.place(x=130, y=270, anchor="center")#se define la colocacion de la ventana de cantidad de  moneda 
        self.capital.place(x=130, y=70, anchor="center")#se define la colocacion de la ventana de moneda origen
        self.interes.place(x=130, y=170, anchor="center")#se define la colocacion de la ventana de moneda destino
        
        self.calcular_button.place(x=430, y=350, anchor="center")#se define la colocacion del boton de convertir
        

        self.pantalla_cuota_anual.configure(font=("Arial Black", 18))#se configura el tipo de letra y tamaño de la ventana de cantidad de  moneda 
        self.pantalla_cuota_mensual.configure(font=("Arial Black", 18))#se configura el tipo de letra y tamaño de la ventana de cantidad de  moneda 
        self.duracion_prestamo.configure(font=("Arial Black", 18))#se configura el tipo de letra y tamaño de la ventana de cantidad de  moneda 
        self.calcular_button.configure(font=("Arial", 14))#se configura el tipo de letra y tamaño del boton de convertir
        self.capital.configure(font=("Arial Black", 18))#se configura el tipo de letra y tamaño de la ventana de cantidad de  moneda 
        self.interes.configure(font=("Arial Black", 18))#se configura el tipo de letra y tamaño de la ventana de cantidad de  moneda 


        etiqueta_cuota_anual= tk.Label(self, text="Cuota anual:", fg="#191007",bg="#FFC06E", font=("Arial", 13))#se define la etiqueta de la pantalla de resultados
        etiqueta_cuota_anual.place(x=430, y=50, anchor="center")#se define la colocacion de la etiqueta de la pantalla de resultado
        etiqueta_cuota_mensual= tk.Label(self, text="Cuota mensual:", fg="#191007",bg="#FFC06E", font=("Arial", 13))#se define la etiqueta de la pantalla de resultados
        etiqueta_cuota_mensual.place(x=430, y=150, anchor="center")#se define la colocacion de la etiqueta de la pantalla de resultado
        

        duracion_prestamo_label = tk.Label(self, text="Plazo en años:", fg="#191007",bg="#82715A", font=("Arial", 13) )#se define la etiqueta de cantidad de  moneda 
        duracion_prestamo_label.place(x=130, y=230, anchor="center")#se define la colocacion de la etiqueta de cantidad moneda 
        

        capital_label = tk.Label(self, text="Capital:", fg="#191007",bg="#82715A", font=("Arial", 13))#se define la etiqueta de moneda origen
        capital_label.place(x=130, y=30, anchor="center")#se define la colocacion de la etiqueta de moneda origen

        interes_label = tk.Label(self, text="Interés:", fg="#191007",bg="#82715A", font=("Arial", 13))#se define la etiqueta de moneda destino
        interes_label.place(x=130, y=130, anchor="center") #se define la colocacion de la etiqueta de moneda destino
        calcular_button_label = tk.Label(self, text=""" 
    Calculadora de préstamos
                    
Introduzca la cantidad de capital, el interés a aplicar 
y el número de años para su amortización 
Con el botón Calcular obtendrá la cuota mensual 
              y la cuota anual resultantes.                                   

        CC-BY-NC 2023 Juan Fuente         
                        """, fg="#191007", bg="#82715A", font=("Arial", 13),)
        calcular_button_label.place(x=275, y=520, anchor="center", width= 500) #se define la colocacion de la etiqueta bajo el boton de convertir
        

        

    def calcular_cuota(self):
        #cantidad = self.duracion_prestamo.get()
        capital = self.capital.get()
        interes = self.interes.get()
        duracion_prestamo = self.duracion_prestamo.get()

        if not capital.replace('.', '', 1).isdigit():
            self.capital.delete(0, tk.END)
            self.pantalla_cuota_mensual.delete(0, tk.END)
            self.pantalla_cuota_anual.delete(0, tk.END)
            messagebox.showerror("Error", "El valor del capital no es válido")
            return
        if not interes.replace('.', '', 1).isdigit():
            self.interes.delete(0, tk.END)
            self.pantalla_cuota_mensual.delete(0, tk.END)
            self.pantalla_cuota_anual.delete(0, tk.END)
            messagebox.showerror("Error", "El valor del interés no es válido")
            return
        #if not 1 <= float(duracion_prestamo) <= 35 or not duracion_prestamo.isdigit():
        if not duracion_prestamo.isdigit():

            self.duracion_prestamo.delete(0, tk.END)
            self.pantalla_cuota_mensual.delete(0, tk.END)
            self.pantalla_cuota_anual.delete(0, tk.END)
            messagebox.showerror("Error", "El valor de la duración del préstamo no es válido")
            return
        
        duracion_meses = float(duracion_prestamo) * 12
        capital, interes, duracion_prestamo = float(capital), float(interes), float(duracion_prestamo)
        cuota_anual = round(((capital * interes / 100) / (1 - (1 + interes / 100) ** -duracion_prestamo)), 2)
        cuota_mensual = round(((capital * interes / 1200) / (1 - (1 + interes / 1200) ** -duracion_meses)), 2)
        return cuota_anual, cuota_mensual
        """except ValueError:
            
            return None, None"""

        
        """ if not nombre_moneda_origen or not nombre_moneda_destino:
            if event is None or (event.widget == self.calcular_button if event else False):
                tk.messagebox.showerror("Error", "Por favor, seleccione la moneda de origen y la moneda de destino.")
                return
"""

    def mostrar_cuota(self):  # New function to handle the calculation and display
        cuota_anual,cuota_mensual = self.calcular_cuota()
        
    
        self.pantalla_cuota_anual.delete(0, tk.END)
        self.pantalla_cuota_anual.insert(tk.END, cuota_anual)

        self.pantalla_cuota_mensual.delete(0, tk.END)
        self.pantalla_cuota_mensual.insert(tk.END, cuota_mensual)
        

if __name__ == "__main__":
    ventana = calculadora_prestamos()
    ventana.mainloop()